core/shared/src/test/scala/fs2/Fs2Spec.scala

```diff
@@ -1,48 +1,114 @@
 package fs2
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
 import scala.concurrent.duration._
 
-import cats.effect.{ContextShift, IO, Timer}
+import cats.{Functor, Monad}
+import cats.effect.{ContextShift, IO, Sync, Timer}
+import cats.implicits._
 
-import org.typelevel.discipline.Laws
-import org.scalatest.{ Args, Assertion, AsyncFreeSpec, FreeSpec, Matchers, Status, Succeeded, Suite }
-import org.scalatest.concurrent.{ AsyncTimeLimitedTests, TimeLimitedTests }
-import org.scalatestplus.scalacheck.{ Checkers, ScalaCheckDrivenPropertyChecks }
+import org.scalatest.{Args, Assertion, Matchers, Status, Succeeded}
+import org.scalatest.check.Checkers
+import org.scalatest.concurrent.AsyncTimeLimitedTests
+import org.scalatest.freespec.AsyncFreeSpec
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
 import org.scalatest.time.Span
 
-abstract class Fs2Spec extends FreeSpec with Fs2SpecLike with TimeLimitedTests with Checkers {
-  val timeLimit: Span = timeout
+import org.typelevel.discipline.Laws
 
-  def checkAll(name: String, ruleSet: Laws#RuleSet): Unit =
-    for ((id, prop) ← ruleSet.all.properties)
-      s"${name}.${id}" in check(prop)
-}
+abstract class Fs2Spec
+    extends AsyncFreeSpec
+    with AsyncTimeLimitedTests
+    with Matchers
+    with GeneratorDrivenPropertyChecks
+    with Checkers
+    with MiscellaneousGenerators
+    with ChunkGenerators
+    with StreamGenerators
+    with EffectTestSupport
+    with TestPlatform {
 
-abstract class AsyncFs2Spec extends AsyncFreeSpec with Fs2SpecLike with AsyncTimeLimitedTests {
+  implicit val timeout: FiniteDuration = 10.seconds
```

Comment:
Bump this back to 60 seconds -- this is likely the cause of most/all of the timeouts on Travis

Reply:
btw, maybe this requires a different PR, but I think we should investigate using `TestContext` more instead of actual time